drivers/rkt/driver_test.go

```diff
@@ -660,24 +661,32 @@ func TestRktDriver_Exec(t *testing.T) {
 	expected := []byte("etcd version")
 	testutil.WaitForResult(func() (bool, error) {
 		res, err := d.ExecTask(task.ID, []string{"/etcd", "--version"}, time.Second)
-		require.NoError(err)
-		require.True(res.ExitResult.Successful())
+		if err != nil {
+			return false, fmt.Errorf("failed to exec: %#v", err)
+		}
+		if !res.ExitResult.Successful() {
+			return false, fmt.Errorf("/etcd --version failed: %#v %#v", res.ExitResult, res)
+		}
 		raw := res.Stdout
 		return bytes.Contains(raw, expected), fmt.Errorf("expected %q but found:\n%s", expected, raw)
 	}, func(err error) {
-		t.Fatalf("err: %v", err)
+		require.NoError(err)
 	})
 
 	// Run command that should fail
 	expected = []byte("flag provided but not defined")
 	testutil.WaitForResult(func() (bool, error) {
 		res, err := d.ExecTask(task.ID, []string{"/etcd", "--cgdfgdfg"}, time.Second)
-		require.False(res.ExitResult.Successful())
-		require.Nil(err)
+		if err != nil {
+			return false, fmt.Errorf("failed to exec: %#v", err)
+		}
+		if res.ExitResult.Successful() {
+			return false, fmt.Errorf("/etcd --cgdfgdfg unexpected succeeded: %#v %#v", res.ExitResult, res)
+		}
 		raw := res.Stdout
 		return bytes.Contains(raw, expected), fmt.Errorf("expected %q but found:\n%s", expected, raw)
 	}, func(err error) {
-		t.Fatalf("err: %v", err)
+		require.NoError(err)
```

Comment:
This func is only called if err is non-nil, so `t.Fatalf` is fine here. `t.Fatal(err)` would probably be the optimal.

Reply:
You are right; though, I much prefer the `require` multi-line formatting of testify that makes it much easier to spot the failing assertion vs a single log line among tons of spurious ones.